src/bci_build/package/bind.py

```diff
@@ -0,0 +1,84 @@
+from pathlib import Path
+
+from bci_build.os_version import ALL_NONBASE_OS_VERSIONS
+from bci_build.os_version import CAN_BE_LATEST_OS_VERSION
+from bci_build.package import DOCKERFILE_RUN
+from bci_build.package import ApplicationStackContainer
+from bci_build.package import ParseVersion
+from bci_build.package import Replacement
+
+_BIND_FILES = {
+    "entrypoint.sh": (
+        (_bind_dir := Path(__file__).parent / "bind") / "entrypoint.sh"
+    ).read_text(),
+    "healthcheck.sh": (_bind_dir / "healthcheck.sh").read_text(),
+}
+
+BIND_CONTAINERS = [
+    ApplicationStackContainer(
+        name="bind",
+        os_version=os_version,
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        version=9,
+        pretty_name="ISC BIND 9",
+        version_in_uid=False,
+        package_list=["bind"],
+        additional_versions=[
+            (_bind_minor_re := "%%bind_major_minor%%"),
+            (_bind_patch_re := "%%bind_major_minor_patch%%"),
+        ],
+        replacements_via_service=[
+            Replacement(_bind_minor_re, "bind", parse_version=ParseVersion.MINOR),
+            Replacement(_bind_patch_re, "bind", parse_version=ParseVersion.PATCH),
+        ],
+        extra_files=_BIND_FILES,
+        env={
+            # copy-pasta from /etc/sysconfig/named
+            "RNDC_KEYSIZE": 512,
+            "NAMED_ARGS": "",
+            "NAMED_INITIALIZE_SCRIPTS": "",
+            # need to set this one so that we can override it
+            "NAMED_CONF": "/etc/named.conf",
+        },
+        custom_end=rf"""COPY entrypoint.sh {(_entrypoint := "/usr/local/bin/entrypoint.sh")}
+COPY healthcheck.sh {(_healthcheck := "/usr/local/bin/healthcheck.sh")}
+{DOCKERFILE_RUN} \
+    chmod +x {_entrypoint}; \
+    chmod +x {_healthcheck};
+
+EXPOSE 53/udp 53/tcp 953/tcp 853/tcp 443/tcp
+
+# patch named.prep to not call logger (provided by systemd)
+# and just log to stdout
+{DOCKERFILE_RUN} \
+    mkdir -p /usr/local/lib/bind; \
+    cp {os_version.libexecdir}bind/named.prep {(_named_prep := "/usr/local/lib/bind/named.prep")}; \
+    sed -i -e 's|logger "Warning: \$1"|echo "Warning: \$1" >\&2|' -e '/\. \$SYSCONFIG_FILE/d' {_named_prep}
+
+# create directories that tmpfiles.d would create for us
```

Comment:
I wonder if there's a way for us to run the tmpfiles.d stuff without having to duplicate it in the dockerfile? this might risk a maintenance burden / incompatibility that we'll eventually miss years down the line.

Reply:
Well, we can install systemd into the container, run systemd-tmpfiles and then remove systemd afterwards, but that sadly leaves a bunch of crap (mostly empty directories and pam backup files, but those are only around 32kb, so I guess we could live with that).

Nevertheless, I have specifically added tests to https://github.com/SUSE/BCI-tests/pull/771 which verify that the systemd-tmpfiles settings have been applied correctly